dkist/dataset/dataset.py

```diff
@@ -121,8 +122,27 @@ def __getitem__(self, item):
         sliced_dataset = super().__getitem__(item)
         if self._file_manager is not None:
             sliced_dataset._file_manager = self._file_manager._slice_by_cube(item)
+            sliced_dataset.meta = sliced_dataset.meta.copy()
```

Comment:
This is stopping changes from the new, sliced, headers from being reflected in the table on the original dataset object. Not sure what to do about that, since we need it to make sure the original header table doesn't get sliced. How does file manager solve this problem?